src/components/form/SeoSettingsBlock.tsx

```diff
@@ -4,41 +4,50 @@ import { useAuth } from 'lodestar-app-element/src/contexts/AuthContext'
 import { handleError } from 'lodestar-app-element/src/helpers'
 import React, { useState } from 'react'
 import { useIntl } from 'react-intl'
-import { AdminBlock, AdminBlockTitle } from '../../components/admin'
-import { commonMessages, craftPageMessages } from '../../helpers/translation'
-import { useMutateAppPage } from '../../hooks/appPage'
-import { CraftPageAdminProps } from '../../types/craft'
+import { commonMessages } from '../../helpers/translation'
+import { MetaTag } from '../../types/general'
+import { AdminBlock, AdminBlockTitle } from '../admin'
 
 type FieldProps = {
   pageTitle: string
   description: string
   keywords: string
 }
 
-const CraftPageSeoSettingBlock: React.VFC<{
-  pageAdmin: CraftPageAdminProps | null
+const SeoSettingsBlock: React.VFC<{
+  id?: string
+  metaTags?: MetaTag | null
+  updateMetaTag: (options?: any) => Promise<any>
   onRefetch?: () => void
-}> = ({ pageAdmin, onRefetch }) => {
+}> = ({ id, metaTags, updateMetaTag, onRefetch }) => {
   const { currentMemberId } = useAuth()
   const { formatMessage } = useIntl()
-  const { updateAppPage } = useMutateAppPage()
   const [form] = useForm<FieldProps>()
   const [loading, setLoading] = useState(false)
 
-  if (!pageAdmin) {
+  if (!id) {
     return <Skeleton active />
   }
 
   const handleSubmit = (values: FieldProps) => {
     if (!currentMemberId) {
       return
     }
+    if (!id) {
+      message.error(formatMessage(commonMessages.event.failedSave))
+      setLoading(false)
+      return
+    }
+
     setLoading(true)
-    updateAppPage({
-      pageId: pageAdmin.id,
-      metaTags: {
-        ...pageAdmin.metaTags,
-        seo: { pageTitle: values.pageTitle, description: values.description, keywords: values.keywords },
+
+    updateMetaTag({
+      variables: {
+        id: id,
+        metaTags: {
+          ...metaTags,
+          seo: { pageTitle: values.pageTitle, description: values.description, keywords: values.keywords },
+        },
```

Comment:
another pattern
use Handling Events onSubmit
passing out event
and update outside
pure component
just mention.... dont need to change